milli/src/update/index_documents/mod.rs

```diff
@@ -282,16 +279,62 @@ where
         let index_documents_ids = self.index.documents_ids(self.wtxn)?;
         let index_is_empty = index_documents_ids.len() == 0;
         let mut final_documents_ids = RoaringBitmap::new();
+        let mut word_pair_proximity_docids = Vec::new();
+        let mut word_position_docids = Vec::new();
+        let mut word_docids = Vec::new();
 
         let mut databases_seen = 0;
         (self.progress)(UpdateIndexingStep::MergeDataIntoFinalDatabase {
             databases_seen,
             total_databases: TOTAL_POSTING_DATABASE_COUNT,
         });
 
-        for typed_chunk in lmdb_writer_rx {
+        for result in lmdb_writer_rx {
+            let typed_chunk = match result? {
+                TypedChunk::WordDocids(chunk) => {
+                    // We extract and mmap our chunk file to be able to get it for next processes.
+                    let mut file = chunk.into_inner();
+                    let mmap = unsafe { memmap2::Mmap::map(&file)? };
+                    let cursor_mmap = CursorClonableMmap::new(ClonableMmap::from(mmap));
+                    let chunk = grenad::Reader::new(cursor_mmap)?;
+                    word_docids.push(chunk);
+
+                    // We reconstruct our typed-chunk back.
+                    file.rewind()?;
+                    let chunk = grenad::Reader::new(file)?;
```

Comment:
could you factorize this in a function?
Is [into_clonable_grenad](https://github.com/meilisearch/milli/blob/main/milli/src/update/index_documents/helpers/grenad_helpers.rs#L69-L77) could suit this case?

Reply:
Done by bumping the grenad version to v0.4.1 and then changing the behavior of the `into_cloneable_grenad` (renamed `as_cloneable_grenad`) by naming it take a reference of a `grenad::Reader`. Doing so reduced the complexity of this quoted code part.